### PR TITLE
feat: standardize documentation sections and add Enterprise Architect…

### PR DESCRIPTION
**Pull request by Anselmoo**
… tooling

- Replace "References" with "Further Reading" across all prompt builders and tests
  - Add disclaimer about external references in buildFurtherReadingSection
  - Update test assertions to match new section naming
  - Ensure backward compatibility and edge case handling

- Add Enterprise Architect Prompt Builder (2025 Enhanced Edition)
  - Implement buildFurtherReadingSection with multiple input format support
  - Add comprehensive test coverage for new builder
  - Include demo script for Enterprise Architect workflows

- Improve tooling and documentation
  - Add markdown link check scripts to package.json
  - Update .lycheeignore for better pattern clarity
  - Enhance demo file formatting and input path clarity
  - Add detailed comments explaining further reading resources

All tests passing (30/30). No breaking changes to public APIs.

## Summary
Briefly explain the changes and why they are needed.

## Changes
- [ ] Code
- [ ] Tests
- [ ] Docs
- [ ] CI/CD

## Checklist
- [ ] Biome checks pass (`npm run check`)
- [ ] TypeScript passes (`npm run type-check`)
- [ ] Tests pass locally (`npm run test:all`)
- [ ] Updated README/docs as needed

## Related Issues
Fixes #

## Notes
Anything reviewers should know.

### CODE REVIEW

File: src/tools/guidelines-validator.ts [hunk 1]
```diff
@@ -185,32 +189,76 @@ function validateAgainstGuidelines(
...
+			{
+				title: "AI-Assisted Sprint Planning Tools 2025",
+				url: "https://www.zenhub.com/blog-posts/the-7-best-ai-assisted-sprint-planning-tools-for-agile-teams-in-2025",
+				description: "ZenHub's guide to modern sprint planning tools",
+			},
 		],
 	} as const;
 	const set = [...(byCat[category as keyof typeof byCat] || []), ...common];
-	return asList ? set : set.join("\n");
+	return asList ? set : JSON.stringify(set);
```
[the remark this answers is not in the document]
Comment (Anselmoo) [reply src/tools/guidelines-validator.ts]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/Anselmoo/mcp-ai-agent-guidelines/pull/189#discussion_r2470787448)

**Comment by Anselmoo** [comment]
## 🔧 Fix for Bot Blocking CI Workflow

### Problem
The `github-actions[bot]` commits are causing the "Lint & Code Quality (Lefthook)" check to remain in a pending state (orange, waiting) because **commits made using `GITHUB_TOKEN` don't trigger workflow runs**. This is a GitHub security feature to prevent infinite workflow loops.

### Root Cause
When the auto-regenerate-demos workflow commits demo files, it uses the default `GITHUB_TOKEN`. GitHub intentionally blocks these commits from triggering other workflows, so the lefthook quality gate check never starts for the bot's commits.

### Solution Implemented
I've updated the `auto-regenerate-demos.yml` workflow with two changes:

1. **Added `[skip ci]` to bot commits** - Explicitly signals that CI should be skipped for these commits since they only contain auto-generated demo files that don't need linting.

2. **Enhanced PR comment** - The bot now explains that the pending Lefthook check is expected behavior and safe to ignore for demo-only commits.

### What This Means
✅ **Safe to merge** - Demo files are auto-generated from tools and don't require quality checks
✅ **All real code changes** still pass through the full quality gate pipeline
✅ **No security compromise** - The `[skip ci]` only applies to bot commits of generated files

### Alternative Solutions (if needed)
If you want bot commits to trigger workflows, you would need to:
- Create a Personal Access Token (PAT) with repo permissions
- Or use a GitHub App token
- Add it as a repository secret
- Update the workflow to use that token instead of `GITHUB_TOKEN`

However, this is **not recommended** for this use case since:
- Demo files are generated, not hand-written
- They don't need linting or type checking
- It would create unnecessary workflow runs

### Verification
The changes have been committed to the current PR branch (`copilot/sub-pr-189-again`). The next time the auto-regenerate workflow runs, it will include the explanatory comment.
